### PR TITLE
Fix wrong warning message for not well-formed typescript configuration file (close #4154)

### LINKED ISSUE #4154: [Configuration] Fix wrong warning for not well-formed tsconfig.json file
We display for not well-formed configuration file (`.testcaferc.json`, `tsconfig.json`) the error message for hard-coded file name

https://github.com/DevExpress/testcafe/blob/cb184b052d5a56f9baf95497459019810da811ca/src/notifications/warning-message.js#L17

So, if the `tsconfig.json` is not well-formed, we display the message about the `.testcaferc.json` file.

Need to pass the configuration filename for this template.

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

File: src/configuration/configuration-base.ts [hunk 1]
```diff
@@ -45,32 +50,32 @@ export default class Configuration {
...
-                this._overridenOptions.push(key);
+                option.source === OptionSource.Configuration)
+                this._overriddenOptions.push(key);
 
             this._setOptionValue(option, value);
         });
     }
 
-    getOption (key) {
+    public getOption (key: string): OptionValue {
```
Comment (LavrovArtem) [inline src/configuration/configuration-base.ts]: Why we need the `public` keyword?

File: src/configuration/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+export interface FilterOption {
```
Comment (LavrovArtem) [inline src/configuration/interfaces.ts]: Maybe rename as `interfaces.d.ts`

File: src/configuration/option-source.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+enum OptionSource {
+    Configuration = 'Configuration',
+    Input = 'Input'
```
Comment (LavrovArtem) [inline src/configuration/option-source.ts]: Is it necessary to change the first char to upper case?

File: src/configuration/configuration-base.ts (hunk 1 above)
Comment (miherlosev) [reply src/configuration/configuration-base.ts]: This method is used in other modules.
https://github.com/DevExpress/testcafe/blob/9b08199af5841f65540846a186c8d19c78b07d9d/src/index.js#L54

File: src/configuration/interfaces.ts (hunk 2 above)
Comment (miherlosev) [reply src/configuration/interfaces.ts]: At present, the only TestCafe and third-party libraries  definitions stored in files with `.d.ts` extension. So, I guess we don't change the naming style right now.

File: src/configuration/option-source.ts (hunk 3 above)
Comment (miherlosev) [reply src/configuration/option-source.ts]: I've used upper case naming notation as in TypeScript documentation - https://www.typescriptlang.org/docs/handbook/enums.html

**Pull request merged**